block-streamer/src/server.rs

```diff
@@ -0,0 +1,153 @@
+                all_blocks.extend(blocks);
+                transmit_blocks(&all_blocks, &target_addr).await?
+            }
+            Message::ApplicationAPI(ApplicationAPI::TransmitDag { cid, target_addr }) => {
+                let root_block = self.storage.get_block_by_cid(&cid)?;
+                let blocks = self.storage.get_all_blocks_under_cid(&cid)?;
+                let mut all_blocks = vec![root_block];
+                all_blocks.extend(blocks);
+                transmit_blocks(&all_blocks, &target_addr).await?
+            }
+            Message::ApplicationAPI(ApplicationAPI::ImportFile { path }) => {
+                let root_cid = self
+                    .storage
+                    .import_path(&PathBuf::from(path.to_owned()))
+                    .await?;
+                let response = Message::ApplicationAPI(ApplicationAPI::FileImported {
+                    path: path.to_string(),
+                    cid: root_cid.to_string(),
+                });
+                if let Some(sender_addr) = self.sender_addr {
+                    for chunk in self.chunker.chunk(response)? {
+                        self.socket.send_to(&chunk, sender_addr).await?;
+                    }
+                }
+            }
+            Message::ApplicationAPI(ApplicationAPI::ExportDag { cid, path }) => {
+                self.storage.export_cid(&cid, &PathBuf::from(path)).await?
+            }
+            Message::ApplicationAPI(ApplicationAPI::RequestAvailableBlocks) => {
+                let raw_cids = self.storage.list_available_cids()?;
+                let cids = raw_cids
+                    .iter()
+                    .map(|c| c.to_string())
+                    .collect::<Vec<String>>();
+
+                if let Some(sender_addr) = self.sender_addr {
+                    let response =
+                        Message::ApplicationAPI(ApplicationAPI::AvailableBlocks { cids });
+                    for chunk in self.chunker.chunk(response)? {
+                        self.socket.send_to(&chunk, sender_addr).await?;
+                    }
+                }
+            }
+            Message::ApplicationAPI(ApplicationAPI::GetMissingDagBlocks { cid }) => {
+                let missing_blocks = self.storage.get_missing_dag_blocks(&cid)?;
+                if let Some(sender_addr) = self.sender_addr {
+                    let response = Message::ApplicationAPI(ApplicationAPI::MissingDagBlocks {
+                        blocks: missing_blocks,
+                    });
+                    for chunk in self.chunker.chunk(response)? {
+                        self.socket.send_to(&chunk, sender_addr).await?;
+                    }
+                }
+            }
+            Message::DataProtocol(data_msg) => {
+                self.receiver.handle_transmission_msg(data_msg).await?;
+            }
+            message => {
+                info!("Received unhandled message: {:?}", message);
```

Comment:
`error!`?

Reply:
Hmm I see this as less of an error and more of an indicator of not-yet-implemented functionality. Messages that go down this path are valid within the `Message` enum tree, I just haven't built out the handling code yet.

Reply:
Ah, maybe a `todo!` might be more expressive then?

Reply:
Well I don't really want to panic when I receive one of these messages. I added a comment above the case noting that this is the default case for valid but not implemented messages.

Reply:
Thanks for clarifying 🙏